File: mobile/src/main/java/org/openhab/habdroid/background/NfcReceiveActivity.java
package org.openhab.habdroid.background;

import android.app.Activity;
import android.content.Intent;
import android.nfc.NfcAdapter;
import android.os.Build;
import android.os.Bundle;
import android.text.TextUtils;

import org.openhab.habdroid.model.NfcTag;
import org.openhab.habdroid.ui.MainActivity;

public class NfcReceiveActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent intent = getIntent();
        if (intent == null || intent.getData() == null) {
            finish();
            return;
        }

        if (Intent.ACTION_VIEW.equals(intent.getAction())
                || NfcAdapter.ACTION_NDEF_DISCOVERED.equals(intent.getAction())) {
            NfcTag tag = NfcTag.fromTagData(intent.getData());
            BackgroundTasksManager.enqueueNfcUpdateIfNeeded(this, tag);
            if (tag != null && !TextUtils.isEmpty(tag.sitemap())) {
                Intent startMainIntent = new Intent(this, MainActivity.class);
                startMainIntent.setAction(MainActivity.ACTION_SITEMAP_SELECTED);
                startMainIntent.putExtra(MainActivity.EXTRA_SITEMAP_URL, tag.sitemap());
                startActivity(startMainIntent);
            }
        }

        if (Build.VERSION.SDK_INT >= 21) {
            finishAndRemoveTask();
        } else {
            finish();
        }
    }
}

File: mobile/src/main/java/org/openhab/habdroid/background/NfcReceiveActivity.kt
/*
 * Copyright (c) 2010-2020 Contributors to the openHAB project
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */

package org.openhab.habdroid.background

import android.app.Activity
import android.content.Intent
import android.nfc.NfcAdapter
import android.os.Bundle
import org.openhab.habdroid.model.toTagData
import org.openhab.habdroid.ui.MainActivity

class NfcReceiveActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (intent == null) {
            finishAndRemoveTask()
            return
        }

        if (intent.action == Intent.ACTION_VIEW || intent.action == NfcAdapter.ACTION_NDEF_DISCOVERED) {
            val tag = intent.data?.toTagData()
            val sitemap = tag?.sitemap
            BackgroundTasksManager.enqueueNfcUpdateIfNeeded(this, tag)
            if (!sitemap.isNullOrEmpty()) {
                val startMainIntent = Intent(this, MainActivity::class.java).apply {
                    action = MainActivity.ACTION_SITEMAP_SELECTED
                    putExtra(MainActivity.EXTRA_SITEMAP_URL, sitemap)
                }
                startActivity(startMainIntent)
            }
        }

        finishAndRemoveTask()
    }
}
